### DISCUSSION TITLE
Custom domain for public dashboard

### DISCUSSION THREAD

**Discussion by kasbah** [plausible/analytics#92] [Anything Else]
Would be nice if we could serve up our public dashboard on our own sub-domain without having to self-host it.

**Comment by metmarkosaric** [comment]
Adding more info from [the old roadmap](https://plausible.nolt.io/79):

"It would be cool if publicly available stats was accessible on the custom domain. So that https://plausible.io/example.com could be accessed with e.g. https://stats.example.com. The custom domain root currently returns a 404.

This would be awesome for private stats as well."

**Comment by cassidyjames** [comment]
Came here to ask for exactly this, but the multi-site case makes it a little more complex. We heavily use Plausible at elementary and I have been thinking about setting up some sort of dashboard to make it easier for folks to find all of our stats pages, basically what I can see when logged into our account and on our dashboard. So a custom domain would need to be associated with an org/account, not just a site. And then it could show all of the public sites for that org.

For extra context, we use `stats.elementary.io` as the custom Plausible domain for _all_ of our sites, including e.g. `elementary.io`, `blog.elementary.io`, `builds.elementary.io`, etc.

**Comment by ruurtjan** [comment]
Jep, same here! I wouldn't mind hosting the javascript on a different custom domain as the dashboard though, if that makes it easier to incorporate @cassidyjames's additional request.

**Comment by KristerV** [comment]
This doesn't seem to have a lot of votes, but me and my company could sure use this.

I'm planning to dive in the code and make the change. But before I commit, @ukutaht wanna give a few guidelines about how you want this done? Assuming you've thought about it. If not I'll figure it out myself.

**Reply by ukutaht** [reply]
I think we need to collaborate closer on this because the custom domain infrastructure is not included in this repo. When using custom domains, all traffic is routed through `custom.plausible.io` which is an nginx web server we control. I'm positive that changes would need to be made in our production nginx server.

Our custom domain server is configured to allow any JS filename which will get loaded as the `plausible.js` script. So a request to `yourcustomdomain.com/js/index.js` gets rewritten as `yourcustomdomain.com/js/plausible.js`. We would probably need to move dashboard assets out of `/js/app.js` and `/js/dashboard.js` because on this path anything you load will be the tracking script.

This year we are planning a complete overhaul of the custom domain infrastructure. We will not be managing our own custom domain server anymore, instead we're looking to partner with a CDN provider for this. So I am also a bit worried about doing a bunch of work here that will be thrown away.

**Reply by KristerV** [reply]
> I'm positive that changes would need to be made in our production nginx server.

Yeah I was thinking about this and assuming the same. 

> So I am also a bit worried about doing a bunch of work here that will be thrown away.

Ok, in this case don't waste time, I can wait.

**Reply by webdevelukas** [reply]
@ukutaht is there an update on this topic? :-)
